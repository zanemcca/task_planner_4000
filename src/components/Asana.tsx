import React from 'react';
import { Card, Row, Button } from 'antd';
import { map } from 'lodash';

import '../styles/Asana.css';

import { ITask } from './Task';
import logo from '../asana_logo.png';
import AsanaTask from './AsanaTask';
import { client } from '../lib/asana';
import { useAsanaToken } from '../hooks/auth';

export interface IAsanaProps {
  project?: string
}

// TODO Fetch tasks
const tasks: ITask[] = [{
  description: 'Support search asana tasks from Command Palette',
}, {
  description: 'Add pagination to Asana integration',
}]

const Asana = (props: IAsanaProps) => {
  const [token] = useAsanaToken();
  // TODO Setup the asana client
  // TODO fetch the asana tasks
  return (
    <Card className="Asana">
      <Row className="Asana-title">Asana</Row>
      {!token && <Row className="Asana-connect">
        <Button href={client.app.asanaAuthorizeUrl()} className="Asana-connect-button">
          <img src={logo} className="Asana-logo" height='32px' alt=''/> Connect Asana
        </Button>
      </Row>}
    {token && (
      <React.Fragment>
        <Row className="Asana-project">{
          props.project
        }</Row>
        {map(tasks, task => (
          <Row key={task.description}>
            <AsanaTask {...task}/>
          </Row>
        ))}
      </React.Fragment>
    )}
    </Card>
  );
}

export default Asana;
