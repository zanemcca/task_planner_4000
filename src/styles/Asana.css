.Asana {
  border: 0px;
  background: inherit;
}

.Asana-title {
  font-family: Avenir;
  font-size: 24px;
  line-height: 33px;
  text-align: left;
  color: #333333;
}

.Asana-project {
  font-family: Avenir;
  font-size: 14px;
  line-height: 17px;
  text-align: left;
  margin-bottom: 16px;
  color: #787878;
}

.Asana-connect {
  margin-top: 33px;
}

.Asana-connect-button {
 height: auto;
}

.Asana-logo {
  padding: 5px 7px 5px 0;
}

.Asana-loading {
  text-align: center;
}
