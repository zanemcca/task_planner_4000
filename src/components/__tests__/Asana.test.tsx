import renderer from 'react-test-renderer';

import Asana, { IAsanaProps } from '../Asana';
import moment from 'moment';
import React from 'react';

it('should find and render tasks when project is given', () => {
  const props: IAsanaProps = {
    project: 'test project'
  }
  const component = renderer.create( <Asana {...props}/>)
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})

it('should render a connect to asana button if there is no project given', () => {
  const component = renderer.create( <Asana />)
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})
