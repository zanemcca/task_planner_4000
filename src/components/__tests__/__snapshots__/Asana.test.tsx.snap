// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should find and render tasks when project is given 1`] = `
<div
  className="ant-card Asana ant-card-bordered"
>
  <div
    className="ant-card-body"
    style={Object {}}
  >
    <div
      className="ant-row Asana-title"
      style={Object {}}
    >
      Asana
    </div>
    <div
      className="ant-row Asana-project"
      style={Object {}}
    >
      test project
    </div>
    <div
      className="ant-row"
      style={Object {}}
    >
      <div
        className="ant-card Task ant-card-bordered"
      >
        <div
          className="ant-card-body"
          style={Object {}}
        >
          <div
            className="ant-row Task-header"
            style={Object {}}
          >
            <div
              className="Task-description"
            >
              Support search asana tasks from Command Palette
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="ant-row"
      style={Object {}}
    >
      <div
        className="ant-card Task ant-card-bordered"
      >
        <div
          className="ant-card-body"
          style={Object {}}
        >
          <div
            className="ant-row Task-header"
            style={Object {}}
          >
            <div
              className="Task-description"
            >
              Add pagination to Asana integration
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`should render a connect to asana button if there is no project given 1`] = `
<div
  className="ant-card Asana ant-card-bordered"
>
  <div
    className="ant-card-body"
    style={Object {}}
  >
    <div
      className="ant-row Asana-title"
      style={Object {}}
    >
      Asana
    </div>
    <div
      className="ant-row Asana-connect"
      style={Object {}}
    >
      <button
        className="ant-btn Asana-connect-button"
        onClick={[Function]}
        type="button"
      >
        <img
          alt=""
          className="Asana-logo"
          height="32px"
          src="asana_logo.png"
        />
        <span>
           Connect Asana
        </span>
      </button>
    </div>
    <div
      className="ant-row"
      style={Object {}}
    >
      <div
        className="ant-card Task ant-card-bordered"
      >
        <div
          className="ant-card-body"
          style={Object {}}
        >
          <div
            className="ant-row Task-header"
            style={Object {}}
          >
            <div
              className="Task-description"
            >
              Support search asana tasks from Command Palette
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="ant-row"
      style={Object {}}
    >
      <div
        className="ant-card Task ant-card-bordered"
      >
        <div
          className="ant-card-body"
          style={Object {}}
        >
          <div
            className="ant-row Task-header"
            style={Object {}}
          >
            <div
              className="Task-description"
            >
              Add pagination to Asana integration
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
